src/symmetric_ops.jl

```diff
@@ -6,13 +6,13 @@
 # Double contraction #
 ######################
 
-@gen_code function dcontract{dim, T1, T2}(S1::SymmetricTensor{2, dim, T1}, S2::SymmetricTensor{2, dim, T2})
+@gen_code function dcontract{dim, T1, T2, M}(S1::SymmetricTensor{2, dim, T1, M}, S2::SymmetricTensor{2, dim, T2, M})
```

Comment:
adding this `M` was all that was needed instead of `n_independent_components` some lines below heh